### PR TITLE
Sort hud indicators by their descriptions

### LINKED ISSUE #785: HUD indicators are sorted nonintuitively
First reported by @VigiFyre as part of #548:

> Seems like indicator sorting on screen is based on entity IDs, which doesn't really work well. I placed indicators on the screenshot from left to right, but their order on screen doesn't match placing order because of EntIDs. Would be nice if they were sorted by placement order or alphabet.
> A screenshot depicting HUD indicators sorted based on no apparent ordering.

### PR DESCRIPTION
**Pull request by Vurv78**
Fixes #785

### CODE REVIEW

File: lua/entities/gmod_wire_hudindicator/cl_init.lua [hunk 1]
```diff
@@ -49,80 +49,88 @@ local function DrawHUDIndicators()
...
-						surface.DrawTexturedRect(hudx + 8, currenty + 8, 64, 32)
-
-						local startx = hudx + 40
-						local starty = currenty + 39
-						surface.SetDrawColor(0, 0, 0, 255)
-						surface.DrawLine(startx, starty, startx + indinfo.LineX, starty + indinfo.LineY)
-
-						-- Now the text
-						draw.SimpleText(txt, "Default", hudx + (indinfo.BoxWidth / 2), currenty + 40 + (pbarheight / 2), Color(255, 255, 255, 255), 1, 1)
+	for _, data in ipairs(sorted) do
```
Comment (thegrb93) [inline lua/entities/gmod_wire_hudindicator/cl_init.lua:63]: Would https://wiki.facepunch.com/gmod/Global.SortedPairsByMemberValue work?

Comment (Vurv78) [reply lua/entities/gmod_wire_hudindicator/cl_init.lua:63]: That mutates the target table and doesn't work on non sequential tables

Comment (thegrb93) [reply lua/entities/gmod_wire_hudindicator/cl_init.lua:63]: Mutates the target table? oof

**Review by thegrb93 — APPROVED** [review]

File: lua/entities/gmod_wire_hudindicator/cl_init.lua (hunk 1 above)
Comment (Vurv78) [reply lua/entities/gmod_wire_hudindicator/cl_init.lua]: Nevermind you are completely correct my bad, I saw it on mobile and incorrectly assumed you sent https://wiki.facepunch.com/gmod/table.SortByMember, I didn't think to check if there were globals for special versions of pairs

Comment (thegrb93) [reply lua/entities/gmod_wire_hudindicator/cl_init.lua]: Ah nice, glad you double checked it. I just glanced at the source of that function and misread it so thought you were right about it. Giving it a closer look, looks like it does work as intended.

**Pull request merged**